# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: uriloader/preload/PreloadHashKey.h
# Commit: b8ecf23519c9
# Full Hash: b8ecf23519c9ab5180d08f00c712674bfe968dac
# Author: Yulia Startsev <ystartsev@mozilla.com>
# Date: 2022-02-21 21:39:46
# Regressor Bug: 1742437
# File Overlap Count: 5
# Description:
#   Bug 1742437 - re-scope all generic module/script loader files under js/loader; r=jonco
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D138133
# ==============================================================================

diff -r 85fd9fc9cffa -r b8ecf23519c9 uriloader/preload/PreloadHashKey.h
--- a/uriloader/preload/PreloadHashKey.h	Mon Feb 21 15:39:08 2022 +0000
+++ b/uriloader/preload/PreloadHashKey.h	Mon Feb 21 15:39:09 2022 +0000
@@ -7,11 +7,15 @@
 
 #include "mozilla/CORSMode.h"
 #include "mozilla/css/SheetParsingMode.h"
-#include "mozilla/dom/ScriptKind.h"
+#include "js/loader/ScriptKind.h"
 #include "nsURIHashKey.h"
 
 class nsIPrincipal;
 
+namespace JS::loader {
+enum class ScriptKind;
+}
+
 namespace mozilla {
 
 namespace css {
@@ -44,7 +48,7 @@
 
   // Construct key for "script"
   static PreloadHashKey CreateAsScript(nsIURI* aURI, CORSMode aCORSMode,
-                                       dom::ScriptKind aScriptKind);
+                                       JS::loader::ScriptKind aScriptKind);
   static PreloadHashKey CreateAsScript(nsIURI* aURI,
                                        const nsAString& aCrossOrigin,
                                        const nsAString& aType);
@@ -92,7 +96,7 @@
   nsCOMPtr<nsIPrincipal> mPrincipal;
 
   struct {
-    dom::ScriptKind mScriptKind = dom::ScriptKind::eClassic;
+    JS::loader::ScriptKind mScriptKind = JS::loader::ScriptKind::eClassic;
   } mScript;
 
   struct {
